### PR TITLE
Fixes syndicate monkey factions, silicon syndicate turret checks

### LINKED ISSUE #36256: Syndie Lavaland Base Turrets Shoot Monkeys
Server-Side Code Modifications: https://tgstation13.org/parsed-logs/basil/server_side_modifications.dm
Round ID: 84647
Server revision compiled on: 2018-03-08
The following pull requests are currently test merged:
#36186: 'this must be the work of a syndicate guardian: adds a new gamemode' by Iamgoofball at commit 0a4f9a
Based off origin/master commit: 84d4b3
Revision: 79b96dd55df6140a0ea79a6c62b9af869a655c9e
Server tools version: /tg/station 13 Server v3.2.1.15
Server tools API version: 3.2.0.2
Current Informational Settings:
Protect Authority Roles From Traitor: 1
Protect Assistant Role From Traitor: 0
Enforce Human Authority: 1
Allow Latejoin Antagonists: 1
Enforce Continuous Rounds: 8 of 19 roundtypes
Allow Midround Antagonists: 6 of 19 roundtypes
All Game Mode Odds:
traitorchan 24.5%
traitor 18.4%
clockwork_cult 8.2%
cult 8.2%
secret_extended 2%
nuclear 16.3%
revolution 12.2%
wizard 10.2%

Easy to see, just pull a monkey out and the turrets shoot it. Admin said they already had Syndie faction.

### PR DESCRIPTION
**Pull request by ShizCalev**
Closes #36374
Fixes #36256

### CODE REVIEW

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -381,7 +381,7 @@
 
 				if(iscyborg(sillycone))
 					var/mob/living/silicon/robot/sillyconerobot = A
-					if(faction == "syndicate" && sillyconerobot.emagged == 1)
+					if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == 1)
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret.dm]: This wasn't fixed when turret factions were made into lists.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 2]
```diff
@@ -381,7 +381,7 @@
 
 				if(iscyborg(sillycone))
 					var/mob/living/silicon/robot/sillyconerobot = A
-					if(faction == "syndicate" && sillyconerobot.emagged == 1)
+					if(LAZYLEN(faction) && (ROLE_SYNDICATE in faction) && sillyconerobot.emagged == TRUE)
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret.dm]: This wasn't fixed when turret factions were made into lists.

**Review by Cyberboss — APPROVED** [review]

**Comment by WJohn** [comment]
aw fuck i didn't even see this

**Comment by ExcessiveUseOfCobblestone** [comment]
Good change but still think we should try to lowercase those strings so they're consistent (and things like this don't happen in the future since it's a standard).

**Pull request merged**